lms/djangoapps/lti_provider/tests/test_views.py

```diff
@@ -0,0 +1,183 @@
+"""
+Tests for the LTI provider views
+"""
+
+from django.conf import settings
+from django.test import TestCase
+from django.test.client import RequestFactory
+from mock import patch, MagicMock
+
+from lti_provider import views
+from lti_provider.signature_validator import SignatureValidator
+from student.tests.factories import UserFactory
+
+
+def set_lti_parameters(dictionary):
+    """
+    Helper method to set up a valid LTI launch.
+    """
+    dictionary['roles'] = u'Instructor,urn:lti:instrole:ims/lis/Administrator'
+    dictionary['context_id'] = u'lti_launch_context_id'
+    dictionary['oauth_version'] = u'1.0'
+    dictionary['oauth_consumer_key'] = u'consumer_key'
+    dictionary['oauth_signature'] = u'OAuth Signature'
+    dictionary['oauth_signature_method'] = u'HMAC-SHA1'
+    dictionary['oauth_timestamp'] = u'OAuth Timestamp'
+    dictionary['oauth_nonce'] = u'OAuth Nonce'
```

Comment:
Dicts (and Django's QueryDicts) support an `update()` call. So you could instead do this:

```python
LTI_DEFAULT_PARAMS = {
    "roles": u'Instructor,urn:lti:instrole:ims/lis/Administrator',
    "context_id": u'lti_launch_context_id',
    # etc...
}
```

Wherever you want to add them to an existing dict like `request.POST`, you could do:

```python
request.POST.update(LTI_DEFAULT_PARAMS)
```